R/favorites.R

```diff
@@ -42,133 +37,78 @@ get_favorites <- function(user,
-    rt <- Map(
-      get_favorites_,
-      user = user,
-      n = n,
-      since_id = since_id,
-      max_id = max_id,
-      parse = parse,
-      token = list(token)
-    )
-  } else if (!is.null(since_id)) {
-    rt <- Map(
-      get_favorites_,
-      user = user,
-      n = n,
-      since_id = since_id,
-      parse = parse,
-      token = list(token)
-    )
-  } else if (!is.null(max_id)) {
-    rt <- Map(
-      get_favorites_,
-      user = user,
-      n = n,
-      max_id = max_id,
-      parse = parse,
-      token = list(token)
-    )
-  } else {
-    rt <- Map(
-      get_favorites_,
-      user = user,
-      n = n,
-      parse = parse,
-      token = list(token)
-    )
-  }
   if (parse) {
-    nms <- all_uq_names(rt)
-    for (i in seq_along(rt)) {
-      for (j in nms) {
-        if (!has_name_(rt[[i]], j) && nrow(rt[[i]]) > 0L) {
-          rt[[i]][[j]] <- NA
-          if (j == "created_at") {
-            rt[[i]][[j]] <- as.POSIXct(rt[[i]][[j]])
-          }
-        } else if (!has_name_(rt[[i]], j) && nrow(rt[[i]]) == 0L) {
-          rt[[i]] <- tibble::as_tibble(data.frame(j = NA))
-          names(rt[[i]]) <- j
-          if (j == "created_at") {
-            rt[[i]][[j]] <- as.POSIXct(rt[[i]][[j]])
-          }
-        }
-      }
-    }
-    ## add favoriter variable to data frames
-    for (i in seq_along(user)) {
-      if (nrow(rt[[i]]) > 0) {
-        rt[[i]]$favorited_by <- user[i]
-      }
-    }
     rt <- do_call_rbind(rt)
   }
   rt
 }
 
-get_favorites_ <- function(user,
+get_favorites_user <- function(user,
                            n = 200,
                            since_id = NULL,
                            max_id = NULL,
                            parse = TRUE,
                            token = NULL) {
-  query <- "favorites/list"
-  if (n > 3000) {
-    warning(paste0("n exceeds max favs returned per ",
-      "token. Setting n to 3000..."),
-      call. = FALSE)
-    n <- 3000
-    count <- 200
-  } else if (n < 200) {
-    count <- n
-  } else {
-    count <- 200
-  }
-  n.times <- rate_limit(token, query)[["remaining"]]
-  if (n.times == 0L) stop("rate limit exceeded", call. = FALSE)
+
   params <- list(
-    user_type = user,
-    count = count,
     tweet_mode = "extended",
     include_ext_alt_text = "true",
     max_id = max_id,
     since_id = since_id
   )
-  names(params)[1] <- .id_type(user)
-  url <- make_url(
-    query = query,
-    param = params)
-  fav <- scroller(url, n, n.times, type = "timeline", token)
+  params[[.id_type(user)]] <- user
+  
+  results <- TWIT_paginate_max_id(token, "favorites/list", params,
+    get_max_id = function(x) x$id_str,
+    page_size = 200,
+    n = n,
+    parse = parse
+  )
+
   if (parse) {
-    fav <- tweets_with_users(fav)
+    results <- tweets_with_users(results)
+    results$favorited_by <- user
+  }
+  
+  results
+}
+
+TWIT_paginate_max_id <- function(token, query, params, 
+                                 get_max_id, 
```

Comment:
A comment about why get_max_id parameter is needed and a explaining why should a function be passed will make maintenance easier, imho.